pkg/node/state_fsm/fsm_common.go

```diff
@@ -22,19 +23,42 @@ func newPeer(fsm FSM, p peer.Peer, peers peer_manager.PeerManager) (FSM, Async,
 	return fsm, nil, nil
 }
 
-func tryBroadcastTransaction(fsm FSM, baseInfo BaseInfo, p peer.Peer, t proto.Transaction) (FSM, Async, error) {
-	t, err := t.Validate(baseInfo.scheme)
-	if err != nil {
+func tryBroadcastTransaction(fsm FSM, baseInfo BaseInfo, p peer.Peer, t proto.Transaction) (_ FSM, _ Async, err error) {
+	defer func() {
+		if err != nil {
+			err = fsm.Errorf(proto.NewInfoMsg(err))
+		}
+	}()
+	if zap.S().Level() <= zap.DebugLevel {
+		defer func() {
+			if genIDErr := t.GenerateID(baseInfo.scheme); genIDErr != nil {
+				zap.S().Errorf("[%s] Failed to generate ID for transaction: %v", fsm.String(), genIDErr)
```

Comment:
Is it okay that we get new ERROR messages if we just enable DEBUG level? Seems more error messages shouldn't appear if we just turn on debug-level.

Reply:
discussed, decided to leave as is.